src/sinks/kafka.rs

```diff
@@ -277,8 +277,12 @@ impl Sink<Event> for KafkaSink {
             "Expected `poll_ready` to be called first."
         );
 
-        let topic = self.topic.render_string(&item).map_err(|missing_keys| {
-            error!(message = "Missing keys for topic.", missing_keys = ?missing_keys);
+        let topic = self.topic.render_string(&item).map_err(|error| {
+            warn!(
+                message = "Failed to render topic template; dropping event.",
+                %error,
+                internal_log_rate_secs = 30,
+            );
```

Comment:
Looks like this and below could use the emit too.

Reply:
Thanks! I misses these components.